### ISSUE TITLE
optim.minimize  error ; invalid argument

### ISSUE THREAD

**Issue by simahgh** [maxpumperla/hyperas#98]
Hi, I have just installed the package but when I am trying to run the simple example in jupyter python, it I gives an error regarding optim.minimize saying invalid argument.
I am wondering if you could help me. Thanks

**Comment by simahgh** [comment]
`
def model(train_X, test_X, train_y, test_y):
    timesteps = 32
    input_dim = 124
    num_classes = 6 
    model = Sequential()
    model.add(LSTM({{choice([256, 120, 80])}}, input_shape=(timesteps, input_dim), return_sequences=True,  activation='tanh'))
    model.add(Dropout({{uniform(0, 1)}}))
    model.add(LSTM(96,activation='tanh',dropout=0.2))
    model.add(Dense(num_classes, activation="softmax"))
    model.compile(loss='categorical_crossentropy', optimizer={{choice(['rmsprop', 'adam', 'sgd'])}})
    model.fit(train_X, train_y,batch_size={{choice([64, 128])}},nb_epoch={{choice([10, 20])}},show_accuracy=True,verbose=0,validation_data=(X_test, Y_test))
    score, acc = model.evaluate(test_X, test_y, show_accuracy=True, verbose=0)
    print('Test accuracy:', acc)
    return {'loss': -acc, 'status': STATUS_OK, 'model': model}

`

**Comment by maxpumperla** [comment]
Can you try to remove the comment in the very line you load the data?

**Comment by simahgh** [comment]
Thanks for your reply.
I did but it still gives some syntax error like:
SyntaxError: from __future__ imports must occur at the beginning of the file

I appreciate your help.

**Comment by maxpumperla** [comment]
yeah, I think in notebooks you can call imports at the beginning of each cell, but 
```
from __future__ import print_function
```
etc. have to be at the beginning of the file in a script, which hyperas translates to. Can you try to move the resp. statements up? If that does not work, the output your command line produces of the hyperas model would be very helpful to see.

**Comment by simahgh** [comment]
Thanks a lot for your reply.
I tried to run the code as you said (put from __future__ import print_function
 first line of first code)but still get the same error:

>>> Imports:
#coding=utf-8

from __future__ import print_function

try:
    from hyperopt import Trials, STATUS_OK, tpe
except:
    pass

try:
    from hyperas import optim
except:
    pass

try:
    from hyperas.distributions import choice, uniform, conditional
except:
    pass

try:
    import scipy.io
except:
    pass

try:
    import numpy as np
except:
    pass

try:
    from scipy.io import loadmat
except:
    pass

try:
    import tensorflow as tf
except:
    pass

try:
    from sklearn.model_selection import train_test_split
except:
    pass

try:
    from sklearn import preprocessing
except:
    pass

try:
    from keras.models import Sequential
except:
    pass

try:
    from keras.layers import Dense
except:
    pass

try:
    from keras.layers import LSTM
except:
    pass

try:
    from keras.layers.core import Dense, Dropout, Activation
except:
    pass

try:
    from keras.layers.embeddings import Embedding
except:
    pass

try:
    from keras.preprocessing import sequence
except:
    pass

try:
    from sklearn import preprocessing
except:
    pass

try:
    from sklearn.preprocessing import OneHotEncoder
except:
    pass

try:
    from sklearn.model_selection import train_test_split
except:
    pass

try:
    import os.path as op
except:
    pass

from __future__ import print_function

try:
    from hyperopt import Trials, STATUS_OK, tpe
except:
    pass

try:
    from hyperas import optim
except:
    pass

try:
    from hyperas.distributions import choice, uniform, conditional
except:
    pass

from __future__ import print_function

try:
    from hyperopt import Trials, STATUS_OK, tpe
except:
    pass

try:
    from hyperas import optim
except:
    pass

try:
    from hyperas.distributions import choice, uniform, conditional
except:
    pass

>>> Hyperas search space:

def get_space():
    return {
        'LSTM': hp.choice('LSTM', [256, 120, 80]),
        'Dropout': hp.uniform('Dropout', 0, 1),
        'optimizer': hp.choice('optimizer', ['rmsprop', 'adam', 'sgd']),
        'batch_size': hp.choice('batch_size', [64, 128]),
        'nb_epoch': hp.choice('nb_epoch', [10, 20]),
    }

>>> Data
 1: 
 2: 
 3: train_X, test_X, train_y, test_y=train_test_split(Xx,Y, test_size=0.1)
 4: 
 5: 
 6: 
>>> Resulting replaced keras model:

  1: def keras_fmin_fnct(space):
  2: 
  3: 
  4:    
  5:     timesteps = 32
  6:     input_dim = 124
  7:     num_classes = 6 
  8:     model = Sequential()
  9:     model.add(LSTM(space['LSTM'], input_shape=(timesteps, input_dim), return_sequences=True,  activation='tanh'))
 10:     model.add(Dropout(space['Dropout']))
 11:     model.add(LSTM(96,activation='tanh',dropout=0.2))
 12:     model.add(Dense(num_classes, activation="softmax"))
 13:     model.compile(loss='categorical_crossentropy', optimizer=space['optimizer'])
 14:     model.fit(train_X, train_y,batch_size=space['batch_size'],nb_epoch=space['nb_epoch'],show_accuracy=True,verbose=0,validation_data=(X_test, Y_test))
 15:     score, acc = model.evaluate(test_X, test_y, show_accuracy=True, verbose=0)
 16:     print('Test accuracy:', acc)
 17:     return {'loss': -acc, 'status': STATUS_OK, 'model': model}
 18: 
Unexpected error: <class 'SyntaxError'>

  File "C:\Users\simah\temp_model.py", line 100
    from __future__ import print_function
                                         ^
SyntaxError: from __future__ imports must occur at the beginning of the file


Thanks a lot for your help

**Comment by simahgh** [comment]
https://gist.github.com/simahgh/15da34b92dffec3d614a5145e0f5fcad
Thanks a lot:)

**Comment by maxpumperla** [comment]
@simahgh it seems hyperas parses the imports in reversed order. How exactly do you invoke python? Can you run examples/simple.py? It has the same import structure as yours and it passes our unit tests.

It's recommended to use `python` or `ipython` from command line.

**Comment by fkolokathi** [comment]
Did you find  a solution for this error.I use Jupyter and I have the same error.
